integration/index.js

```diff
@@ -0,0 +1,159 @@
+import {
+  keys,
+  isUndefined,
+} from 'lodash';
+import request from 'request';
+import deepEqual from 'deep-equal';
+
+const {
+  METAPHYSICS_STAGING_ENDPOINT,
+  METAPHYSICS_PRODUCTION_ENDPOINT,
+} = process.env;
+
+const get = (url, options) => {
+  return new Promise((resolve, reject) => request(url, options, (err, response) => {
+    if (err) return reject(err);
+    resolve(JSON.parse(response.body));
+  }));
+};
+
+const metaphysics = (endpoint) => (query, vars = {}) => {
+  const variables = JSON.stringify(vars);
+  return get(endpoint, { method: 'GET', qs: { query, variables } });
+};
+
+const staging = metaphysics(METAPHYSICS_STAGING_ENDPOINT);
+const production = metaphysics(METAPHYSICS_PRODUCTION_ENDPOINT);
+
+describe('Integration specs', () => {
+  describe('/artwork', () => {
+    const query = `
+      query artwork($id: String!) {
+        artwork(id: $id) {
+          ... banner
+          ... images
+          ... actions
+          ... metadata
+        }
+      }
+      fragment banner on Artwork {
+        banner: related {
+          __typename
+          ... on RelatedSale {
+            name
+            href
+            end_at
+          }
+          ... on RelatedFair {
+            name
+            href
+            profile {
+              icon {
+                img: resized(width: 80, height: 45, version: "square140") {
+                  width
+                  height
+                  url
+                }
+              }
+            }
+          }
+        }
+      }
+      fragment images on Artwork {
+        images {
+          id
+          url(version: "larger")
+          placeholder: resized(width: 30, height: 30, version: "small") {
+            url
+          }
+        }
+      }
+      fragment actions on Artwork {
+        is_shareable
+        is_hangable
+      }
+      fragment metadata on Artwork {
+        href
+        title
+        artists {
+          id
+          name
+          href
+        }
+        medium
+        dimensions {
+          in
+          cm
+        }
+        is_contactable
+        partner {
+          name
+          href
+        }
+      }
+    `;
```

Comment:
I'm thinking that rather than inlining the queries here; we can, with the help of a middleware in force, just fetch the query that constructs a particular page, then run it against prod/staging and compare as below.